### PR TITLE
complex level

### PR DESCRIPTION
**Pull request by bbeck13**
https://github.com/bbeck13/RhythmRunner/issues/44
can you beat it? 
I've been having way to much fun playing this

### CODE REVIEW

**Comment by alexottoboni** [comment]
So this generates different things based on different music files right? It's pretty challenging too. There's some weird visual glitches like if a falling platform is clipped inside the two static platforms next to it it looks like it suddenly expands before dropping, but that's just stuff to fix in the renderer I'm pretty sure.

QA 👍

**Comment by Matthew-Stewart** [comment]
Holding space still OP 💯  QA 🥇

**Comment by bbeck13** [comment]
@alexottoboni yes it does for now you can switch the 2.wav to 1.wav in the main file to use the original song In the future this selection should be in a menu or something. I noticed the dropping platform thing as well it might be a vfc thing but I'm not sure it's not to noticeable anyways.

File: src/game_state/DroppingPlatform.cpp [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+void DroppingPlatform::SetDropping() {
+  dropping = true;
+}
+
+float DroppingPlatform::GetYVelocity() {
+  return dropVel;
+}
+
+void DroppingPlatform::Reset() {
+  position = origionalPosition;
```
Comment (alexottoboni) [inline src/game_state/DroppingPlatform.cpp]: Typo `%s/origional/original/g`

File: src/game_updater/GameUpdater.cpp [hunk 2]
```diff
@@ -10,8 +10,9 @@
 #include "TimingConstants.h"
```
Comment (alexottoboni) [inline src/game_updater/GameUpdater.cpp:10]: Some comment blocks before each type of object is modified might make this easier to read and scan through to make changes later.

**Comment by alexottoboni** [comment]
CR 👍  for this and octtree

**Comment by josepharhar** [comment]
I didn't get a chance to review this, but I just want to say that it really adds the needed interesting gameplay that we were missing. Thanks Braden! :ok_hand: :ok_hand: :100: :100: :eyes: :eyes:

**Pull request merged**